include/utils/utility.h

```diff
@@ -464,6 +465,48 @@ T ReverseBytes::operator() (T & data) const
 }
 
 
+
+/**
+ * Struct which defines a custom comparison object that
+ * can be used with std::sets of std::unique_ptrs
+ */
+template <class T>
```

Comment:
Do we need this to be templated around T?  Or would templating the member functions work?

Reply:
I don't know actually... I just assumed this was how you were supposed to do it, but I can check...

Reply:
OK, I tried templating just the member functions and it still worked. This makes the `typedef` declaration nicer since you don't have to repeat the type twice... so I'll go ahead and make that change.